resources/developer-mode.mjs

```diff
@@ -235,34 +236,44 @@ function createUIForRun() {
     return buttons;
 }
 
-function updateURL() {
-    const url = new URL(window.location.href);
+function updateParamsSuitesAndTags() {
+    params.suites = [];
+    params.tags = [];
 
     // If less than all suites are selected then change the URL "Suites" GET parameter
     // to comma separate only the selected
     const selectedSuites = Suites.filter((suite) => !suite.disabled);
+    if (!selectedSuites.length)
+        return;
+
+    // Try finding common tags that would result in the current suite selection.
+    let commonTags = new Set(selectedSuites[0].tags);
+    for (const suite of Suites) {
+        if (suite.disabled)
+            suite.tags.forEach((tag) => commonTags.delete(tag));
+        else
+            commonTags = new Set(suite.tags.filter((tag) => commonTags.has(tag)));
+    }
+    if (selectedSuites.length > 1 && commonTags.size) {
+        const tags = [...commonTags];
+        if (tags[0] !== "default")
```

Comment:
Implicitly relying on "default" being the first element isn't great. Why not just do this instead?
```js
params.tags = commonTags.has("default") ? [...commonTags] : [];
```

Reply:
ah yes, much nicer that way!